src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ManagedGroupService.scala

```diff
@@ -128,15 +128,27 @@ class ManagedGroupService(private val resourceService: ResourceService, private
   }
 
   def requestAccess(resourceId: ResourceId, requesterUserId: WorkbenchUserId): Future[Unit] = {
-    val resourceAndPolicyName = ResourceAndPolicyName(Resource(ManagedGroupService.managedGroupTypeName, resourceId), ManagedGroupService.adminPolicyName)
-    accessPolicyDAO.listFlattenedPolicyMembers(resourceAndPolicyName).map { users =>
-      val notifications = users.map { recipientUserId =>
-        Notifications.GroupAccessRequestNotification(recipientUserId, WorkbenchGroupName(resourceId.value).value, users, requesterUserId)
-      }
-
-      cloudExtensions.fireAndForgetNotifications(notifications)
+    getAccessInstructions(resourceId).map {
+      case accessInstructions: Some[String] =>
+        throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Please follow special access instructions: $accessInstructions"))
```

Comment:
can you do `Future.failed(new WorkbenchExceptionxxx)`? there was a similar discussion [here](https://github.com/broadinstitute/sam/pull/200#discussion_r215724582)